### ISSUE TITLE
Training results with best epoch on validation set

### ISSUE THREAD

**Issue by czy10383** [ultralytics/ultralytics#20590]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I am training an object detector using YOLOv8. My question is about the validation results on the best.pt presented at the end of the training. I ran my simulations for 4 epochs.

`model = YOLO("yolov8n.pt")
    # Use the model # https://docs.ultralytics.com/usage/cfg/#train-settings
    model.train(data=yamlFile,
                amp=False,
                epochs=4,
                imgsz= 224,
                augment=True,
                batch=16)  # train the model`

Starting training for 4 epochs...
  0%|          | 0/54 [00:00<?, ?it/s]
      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        1/4     0.545G       1.58          2      1.126        256        224: 100%|██████████| 54/54 [00:03<00:00, 14.92it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 3/3 [00:00<00:00,  5.99it/s]
                   all         93       1127      0.769      0.668      0.711      0.331
      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        2/4     0.654G      1.412     0.9125      1.017        285        224: 100%|██████████| 54/54 [00:02<00:00, 19.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 3/3 [00:00<00:00, 11.68it/s]
                   all         93       1127      0.821      0.675      0.756      0.362
  0%|          | 0/54 [00:00<?, ?it/s]
      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        3/4     0.654G      1.336     0.7915     0.9947        235        224: 100%|██████████| 54/54 [00:02<00:00, 19.69it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 3/3 [00:00<00:00, 10.74it/s]
                   all         93       1127      0.784       0.69      0.725      0.345
  0%|          | 0/54 [00:00<?, ?it/s]
      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        4/4     0.654G      1.273      0.725     0.9771        175        224: 100%|██████████| 54/54 [00:03<00:00, 16.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 3/3 [00:00<00:00, 10.00it/s]
                   all         93       1127      0.876      0.738      0.827      0.424

4 epochs completed in 0.006 hours.
Optimizer stripped from runs\detect\train5\weights\last.pt, 6.2MB
Optimizer stripped from runs\detect\train5\weights\best.pt, 6.2MB

Validating runs\detect\train5\weights\best.pt...
Ultralytics 8.3.131  Python-3.10.0 torch-2.3.1+cu121 CUDA:0 (NVIDIA GeForce RTX 4080 Laptop GPU, 12282MiB)
Model summary (fused): 72 layers, 3,005,843 parameters, 0 gradients, 8.1 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 3/3 [00:00<00:00,  3.26it/s]
                   all         93       1127      0.843       0.71      0.807      0.418
Speed: 0.2ms preprocess, 5.2ms inference, 0.0ms loss, 0.9ms postprocess per image

From what i see, the best epoch should be the last epoch, but its results of [0.876      0.738      0.827      0.424] does not match the final validation presented: [0.843       0.71      0.807      0.418]

May I ask why are they not the same?

I have gone through past similar issues: https://github.com/ultralytics/ultralytics/issues/20070

**Comment by UltralyticsAssistant** [comment]
👋 Hello @czy10383, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for additional information on training and validation reporting, as well as reviewing the [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples for further details.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please include as much context as possible, such as dataset examples, training logs, or snippets of your configuration, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best! For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. For in-depth discussions, visit our [Discourse](https://community.ultralytics.com/). Or join threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with other users.

## Upgrade

Upgrade to the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to ensure your issue is not already resolved in the newest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response 🤖. An Ultralytics engineer will review your question and assist you as soon as possible!

**Comment by Y-T-G** [comment]
Remove `augment=True`

**Comment by czy10383** [comment]
Thank you! I have managed to get similar results around the 2nd decimal place , after removing augment=True . I had mistaken augment=True for training set augmentation. Thank you!

**Comment by glenn-jocher** [comment]
The difference in validation metrics occurs because the `augment=True` parameter was activating test-time augmentation (TTA) during validation phases. This causes different results between training validation and final validation.

Just to clarify what happened: the `augment` parameter in YOLO is specifically for test-time augmentation during validation, not for training data augmentation as you initially thought. When enabled, the model validates using multiple transformed versions of each image, which typically produces different metrics than validation without augmentation.

I'm glad removing `augment=True` resolved the issue for you! For consistent validation results, it's best to keep augmentation settings the same throughout your training and evaluation pipeline.
